### DISCUSSION TITLE
A full controlnet

### DISCUSSION THREAD

**Discussion by cibernicola** [lllyasviel/Fooocus#2020] [General]
Are there any plans to implement full controlnet in Fooocus? That would be really interesting.

I thought the main reason was that there were no controlnet models for SDXL, but I see that there are.
In fact the user is .... the same(?) as the Foocus user?

https://huggingface.co/lllyasviel/sd_control_collection/tree/main

**Comment by mashb1t** [comment]
SDXL ControlNet adapters are indeed available for many types and a few have already been implemented in https://github.com/fenneishi/Fooocus-ControlNet-SDXL.
It is not impossible that Fooocus is also going to support these adapters, but i can't say for certain as the core dev team has gone dark and will be available again from end of January.

**Reply by aolko** [reply]
that fork is dead (not upstream)

**Comment by IPv6** [comment]
SDXL controlnets on complex concepts are simply not working well... whats the point of using CN if the generation completely ignore it half of the time? Seems SD structure were more favorable for CNs

would be cool to have SD controlnets on refine stage with SD models though :)

**Comment by mel870** [comment]
I think the main reason why other controlnets were not implemented is because Fooocus was created to be a Midjourney alternative. So to simplify the use, only Canny and Depth models were added, and now IP adapter and Faceswap. I don't think other controlnets will be added since the only SDXL controlnets missing are Recolor, Openpose and Sketch.  You can experiment and get results those controlnet would give.

**Reply by cibernicola** [reply]
I've done some research, believe me, and to be fully impresiveley-dream-capturing, an openponse/dwpose would be really interesting.
